src/adapters/lokijs/worker/workerMock.js

```diff
@@ -32,7 +32,9 @@ export default class LokiWorkerMock {
     this._workerContext = {
       postMessage: data => {
         const clonedData = clone(data)
-        setImmediate(() => {
+
+        // Schedule on the microtask queue to be able to achieve synchronous, glitch-free rendering
+        Promise.resolve().then(() => {
```

Comment:
Hmm wouldn't `requestAnimationFrame` be better? It's guaranteed to run before the next frame while microtasks kinda are more unpredictable running only when stack is empty.

Reply:
Hmm, 

> microtasks kinda are more unpredictable running only when stack is empty.

I don't know anything about that, but my understanding is `rAF` is also unpredictable because spec is not clear about when this should be executed, and if main thread is busy for a long time, it won't be executed immediately, given that its purpose is to schedule… stuff before next animation frame ;) https://stackoverflow.com/a/43050651

I will continue doing research on this and get back on you

Reply:
Yeah I also haven't tested this and my comment was based on things I read/heard. AFAIK `rAF` should run between macrotasks so if 1 task takes a lot of time a frame probably will be skipped. However if thread is blocked by 1 task, microtasks also probably won't run 🤔 But unfortunately this can differ between browsers 🙁